src/frequenz/dispatch/_bg_service.py

```diff
@@ -271,25 +283,41 @@ async def _run(self) -> None:
                             dispatch = Dispatch(selected.message.dispatch)
```

Comment:
It might also add some clarity to rename `dispatch` to `new_dispatch` and use `old_dispatch` for the one we have stored, although is out of the scope of this PR, but mentioning it because it related.

I also strong recommend to use `is None` instead of `not`, as if in the future someone adds a `__bool__` to dispatch, the meaning could change. In general is better practice to use `is None` than implicit conversion to `bool` if you want to check if something returned `None`, we already had several bugs because we used `not x` with `int`/`float` when the value was `0`.

```suggestion
                            new_dispatch = Dispatch(selected.message.dispatch)
                            old_dispatch = self._dispatches.get(new_dispatch.id)
                            is_new_or_newer = (
                                old_dispatch is None
                                or new_dispatch.update_time > old_dispatch.update_time
                            )

```